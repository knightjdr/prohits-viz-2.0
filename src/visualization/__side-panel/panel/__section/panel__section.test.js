import React from 'react';
import { shallow } from 'enzyme';

import PanelSection from './panel__section';

describe('Panel section', () => {
  describe('with border and title', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <PanelSection
          border
          title="test"
        >
          children
        </PanelSection>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have border', () => {
      expect(wrapper.find('.panel__section-border').length).toBe(1);
    });

    it('should have title', () => {
      expect(wrapper.find('.panel__section-title').length).toBe(1);
    });
  });

  describe('without border', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <PanelSection
          border={false}
          title="test"
        >
          children
        </PanelSection>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not have border', () => {
      expect(wrapper.find('.panel__section-border').length).toBe(0);
    });
  });

  describe('without title', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <PanelSection border>
          children
        </PanelSection>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not have title', () => {
      expect(wrapper.find('.panel__section-title').length).toBe(0);
    });
  });
});
