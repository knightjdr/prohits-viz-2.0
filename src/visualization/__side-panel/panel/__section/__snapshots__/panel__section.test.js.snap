// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel section with border and title should match snapshot 1`] = `
<Fragment>
  <div
    className="panel__section-border"
  />
  <div
    className="panel__section-title"
  >
    test
  </div>
  children
</Fragment>
`;

exports[`Panel section without border should match snapshot 1`] = `
<Fragment>
  <div
    className="panel__section-title"
  >
    test
  </div>
  children
</Fragment>
`;

exports[`Panel section without title should match snapshot 1`] = `
<Fragment>
  <div
    className="panel__section-border"
  />
  children
</Fragment>
`;
